### PR TITLE
Result details API

### PR DESCRIPTION
**Pull request by samu80**
This API gives specific candidate results details 
    - passed testcases
    - failed testcases
    - code

### CODE REVIEW

**Review by somnath77 — APPROVED** [review]

File: app/controllers/api/v1/candidate_results_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      def show
+        passed = []
+        failed = []
+        @test_case_results.each do |test_case_result|
+          if test_case_result.is_passed == true
+            passed << test_case_result.test_case
+          else
+            failed << test_case_result.test_case
+          end
+        end
```
Comment (rahulknojha) [inline app/controllers/api/v1/candidate_results_controller.rb:16]: why we are not using active record query.?

File: app/models/submission.rb [hunk 2]
```diff
@@ -6,13 +6,18 @@ class Submission < ApplicationRecord
   belongs_to :problem
   has_many :test_case_results
 
+  scope :get_submissions, lambda { |drives_candidate_id, problem_id|
+    joins(test_case_results: [:test_case]).where(drives_candidate_id: drives_candidate_id,
+                                                 problem_id: problem_id,
+                                                 test_case_results: { is_passed: true }).select('submissions.id as submission_id,
+                                                                                                  sum(test_cases.marks) as marks')
+                                          .group('submissions.id')
+  }
```
Comment (rahulknojha) [inline app/models/submission.rb:15]: indentation ?

**Review by rahulknojha — CHANGES_REQUESTED** [review]

File: app/models/submission.rb (hunk 2 above)
Comment (samu80) [reply app/models/submission.rb:15]: that line is autocorrected by rubocop

File: app/models/submission.rb [hunk 3]
```diff
@@ -6,13 +6,18 @@ class Submission < ApplicationRecord
   belongs_to :problem
   has_many :test_case_results
 
+  scope :get_submissions, lambda { |drives_candidate_id, problem_id|
```
Comment (rahulknojha) [inline app/models/submission.rb:9]: naming convention does not seems proper. what kind of submission we are trying to get is not clear

**Review by rahulknojha — DISMISSED** [review]

**Review by somnath77 — APPROVED** [review]

**Pull request merged**